packages/core/src/utils/SIWXUtil.ts

```diff
@@ -1,4 +1,256 @@
+import { AccountController } from '../controllers/AccountController.js'
+import { W3mFrameRpcConstants } from '@reown/appkit-wallet'
+import { StorageUtil } from './StorageUtil.js'
+
+export const SIWXUtil = {
+  getSIWX() {
+    return OptionsController.state.siwx
+  },
+  async initializeIfEnabled() {
+    const siwx = OptionsController.state.siwx
+    const address = CoreHelperUtil.getPlainAddress(ChainController.getActiveCaipAddress())
+    const network = ChainController.getActiveCaipNetwork()
+
+    if (!(siwx && address && network)) {
+      return
+    }
+
+    try {
+      const sessions = await siwx.getSessions(network.caipNetworkId, address)
+      if (sessions.length) {
+        return
+      }
+
+      await ModalController.open({
+        view: 'SIWXSignMessage'
+      })
+    } catch (error: unknown) {
+      // eslint-disable-next-line no-console
+      console.error('SIWXUtil:initializeIfEnabled', error)
+
+      EventsController.sendEvent({
+        type: 'track',
+        event: 'SIWX_AUTH_ERROR',
+        properties: this.getSIWXEventProperties()
+      })
+
+      // eslint-disable-next-line no-console
+      await ConnectionController._getClient()?.disconnect().catch(console.error)
+      RouterController.reset('Connect')
+      SnackController.showError('A problem occurred while trying initialize authentication')
+    }
+  },
+  async requestSignMessage() {
+    const siwx = OptionsController.state.siwx
+    const address = CoreHelperUtil.getPlainAddress(ChainController.getActiveCaipAddress())
+    const network = ChainController.getActiveCaipNetwork()
+    const client = ConnectionController._getClient()
+
+    if (!(siwx && address && network && client)) {
+      throw new Error('SIWX is not enabled')
+    }
+
+    try {
+      const siwxMessage = await siwx.createMessage({
+        chainId: network.caipNetworkId,
+        accountAddress: address
+      })
+
+      const message = siwxMessage.toString()
+
+      if (StorageUtil.getConnectedConnector() === 'ID_AUTH') {
+        RouterController.pushTransactionStack({
+          view: null,
+          goBack: false,
+          replace: true
+        })
+      }
+
+      const signature = await client.signMessage(message)
+
+      await siwx.addSession({
+        data: siwxMessage,
+        message,
+        signature: signature as `0x${string}`
+      })
+
+      ModalController.close()
+
+      EventsController.sendEvent({
+        type: 'track',
+        event: 'SIWX_AUTH_SUCCESS',
+        properties: this.getSIWXEventProperties()
+      })
+    } catch (error) {
+      const properties = this.getSIWXEventProperties()
+
+      if (properties.isSmartAccount) {
+        SnackController.showError('This application might not support Smart Accounts')
+      } else {
+        SnackController.showError('Signature declined')
+      }
+
+      EventsController.sendEvent({
+        type: 'track',
+        event: 'SIWX_AUTH_ERROR',
```

Comment:
are these events tracked? else we need to notify ata so they start consuming them

Reply:
hmmm I'm not sure, how can I make sure this is tracked or not?